<a href="https://colab.research.google.com/github/jacob-torres/DS-Unit-1-Sprint-2-Statistics/blob/master/module3/Jacob_Torres_LS_DS_123_Confidence_Intervals_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [2]:
#Import statements
import pandas as pd
import numpy as np
from scipy.stats import t


In [3]:
#Print the top 5 rows
coke_url = 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'
coke_df = pd.read_csv(coke_url)
coke_df.head()

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.  Summarize your results in a sentence or two.

In [16]:
#Calculations here
caff_col = coke_df['Caffeine']
mean_caff = caff_col.mean()
std_caff = caff_col.std()
n = caff_col.count()
sterr_caff = std_caff / np.sqrt(n)

print("Mean caffeine content =", mean_caff)
print("Standard deviation =", std_caff)
print("Sample size =", n)
print("Standard error =", sterr_caff)

Mean caffeine content = 37.9402
Standard deviation = 5.243756828216712
Sample size = 50
Standard error = 0.7415792024250598


The average caffeine content for this sample is 37.9mg, with a standard deviation of about 5.2mg and a standard error of about 0.7. The sample size is 50.

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [18]:
#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.
t_star = t.ppf(0.975, df=49)
print('t_star =', t_star)

t_star = 2.009575234489209


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [14]:
margin_err = t_star * sterr_caff
print("Margin of error =", margin_err)

Margin of error = 1.49025919960566


The margin of error for the 95% confidence interval is about 1.5, meaning we can be 95% confident that the mean caffeine content of the population falls within a range of about 1.5 plus or minus the sample mean.

###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [21]:
lower_cl = mean_caff - margin_err
upper_cl = mean_caff + margin_err
ci = (lower_cl, upper_cl)
print(ci)

(36.44994080039434, 39.43045919960566)


###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [20]:
t.interval(alpha=0.95, df=49, loc=mean_caff, scale=sterr_caff)

(36.44994080039434, 39.43045919960566)

###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

The two confidence intervals match at least to the first 3 decimal places.

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

The 95% confidence interval suggests that, with about 95% confidence, we can estimate that the mean caffeine content in the population is between 36.5 and 39.4mg.

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [22]:
# 90% confidence interval
t.interval(alpha=0.9, df=49, loc=mean_caff, scale=sterr_caff)

(36.696904726749196, 39.1834952732508)

The 90% interval calculated above is more narrow, and thus more precise than the 95% interval; 36.7 and 39.2mg, as opposed to 36.5 and 39.4mg. To increase our confidence in the range by that extra 5%, we widen the range by about plus or minus 0.2mg.

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [23]:
#Calculations here
t.interval(alpha=0.99, df=49, loc=mean_caff, scale=sterr_caff)

(35.95280335285685, 39.92759664714315)

The 99% confidence interval is less accurate than the than the 95% confidence interval, with a range of about 35.95 and 39.93.

## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


Answer here.

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

When the sample size is increased to 100, both the standard error and the margin error decrease.

In [26]:
n = 100
sterr_caff = std_caff / np.sqrt(n)
t_star = t.ppf(0.975, df=99)
margin_err = t_star * sterr_caff

print("New sample size =", n)
print("New standard error =", sterr_caff)
print("New margin of error =", margin_err)

New sample size = 100
New standard error = 0.5243756828216712
New margin of error = 1.0404751188137005


Compare MEs here.

